proccon.py

```diff
@@ -0,0 +1,283 @@
+        
+        dot_lines.append("    ")
+        dot_lines.append("    // Parent-child relationships with execution info")
+        
+        # Add parent-child edges with detailed labels
+        edge_count = 0
+        for pid, proc_data in processes.items():
+            ppid = proc_data["ppid"]
+            if ppid in processes:
+                parent_name = processes[ppid]["name"]
+                child_name = proc_data["name"]
+                child_path = proc_data["path"]
+                
+                # Create descriptive edge label
+                edge_label = f"spawned\\n{child_name}"
+                
+                # Add file path if available and different
+                if child_path != child_name and child_path != "N/A":
+                    # Get just filename from path
+                    if '\\' in child_path:
+                        filename = child_path.split('\\')[-1]
+                        if filename != child_name:
+                            edge_label += f"\\n({filename})"
+                
+                dot_lines.append(
+                    f'    proc_{ppid} -> proc_{pid} [label="{edge_label}", color="darkblue", penwidth=2];'
+                )
+                edge_count += 1
+        
+        dot_lines.append("    ")
+        dot_lines.append("    // Legend")
+        dot_lines.append('    subgraph cluster_legend {')
+        dot_lines.append('        label="Legend";')
+        dot_lines.append('        style=filled;')
+        dot_lines.append('        color=lightgrey;')
+        dot_lines.append('        node [shape=box, fontsize=10];')
+        dot_lines.append('        legend_root [label="ROOT\\n(No parent)", fillcolor="lightcoral"];')
+        dot_lines.append('        legend_parent [label="PARENT\\n(Has children)", fillcolor="lightgreen"];')
+        dot_lines.append('        legend_child [label="CHILD/SUBCHILD\\n(Leaf process)", fillcolor="lightblue"];')
+        dot_lines.append('        legend_root -> legend_parent [label="spawned", fontsize=9];')
+        dot_lines.append('        legend_parent -> legend_child [label="spawned", fontsize=9];')
+        dot_lines.append('    }')
+        
+        # Footer
+        dot_lines.append("}")
+        
+        vollog.info(f"Generated graph with {len(processes)} processes and {edge_count} relationships")
+        
+        return "\n".join(dot_lines)
+    
+    def _generator(self):
+        kernel_module_name = self.config["kernel"]
+        output_file = self.config.get("output", "process_tree.dot")
+        
+        vollog.info("Extracting process information from memory dump...")
+        processes = self._get_process_data(kernel_module_name)
+        
+        vollog.info(f"Found {len(processes)} processes")
+        vollog.info("Generating DOT file...")
+        dot_content = self._generate_dot_content(processes)
+        
+        # Write to file
+        try:
+            with open(output_file, 'w', encoding='utf-8') as f:

```

Comment:
Please don't directly write out files.  If this is running as part of a webserver, this will write the data out into the web server, rather than being able to return it to the user.  Instead use the built-in `FileHandlerInterface`, by calling `self.open`.  The parameter it takes is a suggested filename, and will work correctly however the library is called.